pkg/controller/controller_utils.go

```diff
@@ -757,9 +758,8 @@ func (s ActivePods) Less(i, j int) bool {
 		return len(s[i].Spec.NodeName) == 0
 	}
 	// 2. PodPending < PodUnknown < PodRunning
-	m := map[v1.PodPhase]int{v1.PodPending: 0, v1.PodUnknown: 1, v1.PodRunning: 2}
-	if m[s[i].Status.Phase] != m[s[j].Status.Phase] {
-		return m[s[i].Status.Phase] < m[s[j].Status.Phase]
+	if podPhaseRankings[s[i].Status.Phase] != podPhaseRankings[s[j].Status.Phase] {
+		return podPhaseRankings[s[i].Status.Phase] < podPhaseRankings[s[j].Status.Phase]
```

Comment:
you changed the values in the map, how can this be right?  If it's wrong, please add a unit test that will catch it.